tasklist/app/controllers/tasks_controller.rb

```diff
@@ -17,16 +18,40 @@ def delete
 		redirect_to "/"
 	end
 
-	# def edit
-	# 	@task = Task.find(params[:id])
-	# 	render :new
-	# end
-
 	def create
 		Task.create(task_params[:task])
 		redirect_to "/"
 	end
 
+	def edit 
+		show
```

Comment:
Here and in the actions defined below, you're using the `show` action as a way of getting the `@task` variable for the current task.

Because the `show` action might be updated later to do more work that is unrelated to `edit` or `update`, etc. I think it would be best to avoid calling `show` from each of those actions.

Instead, you can use a feature known as a "controller hook", which lets you run a particular method before/after one or more controller actions. In this case I would recommend using the `before_action` hook.
Check out the documentation online, and if you have any questions please ask Jennie as she currently has this working in her project. We'll also be talking about it in class sometime.